packages/application-components/src/components/main-pages/form-main-page/form-main-page.spec.tsx

```diff
@@ -0,0 +1,62 @@
+import Text from '@commercetools-uikit/text';
+import Spacings from '@commercetools-uikit/spacings';
+import { screen, renderComponent } from '../../../test-utils';
+import FormMainPage from './form-main-page';
+
+const Content = () => (
+  <Spacings.Stack>
+    <Text.Body>
+      Lorem ipsum dolor sit amet, consectetur adipiscing elit
+    </Text.Body>
+  </Spacings.Stack>
+);
+
+const renderFormMainPage = (additionalProps = {}) => {
+  renderComponent(
+    <FormMainPage
+      onPrimaryButtonClick={() => {}}
+      onSecondaryButtonClick={() => {}}
+      {...additionalProps}
+    >
+      <Content />
+    </FormMainPage>
+  );
+};
+
+describe('rendering', () => {
+  it('should show form controls and handle button clicks', () => {
+    const mockPrimaryOnClick = jest.fn();
+    const mockSecondaryOnClick = jest.fn();
+
+    renderFormMainPage({
+      title: 'Test title',
+      labelPrimaryButton: 'Test Primary Button',
+      onPrimaryButtonClick: mockPrimaryOnClick,
+      labelSecondaryButton: 'Test Secondary Button',
+      onSecondaryButtonClick: mockSecondaryOnClick,
+    });
+
+    screen.getByText('Test Primary Button').click();
+    expect(mockPrimaryOnClick).toBeCalled();
+
+    screen.getByText('Test Secondary Button').click();
+    expect(mockSecondaryOnClick).toBeCalled();
+  });
+
+  it('should disable form control buttons', () => {
+    renderFormMainPage({
+      title: 'Test title',
+      labelPrimaryButton: 'Test Primary Button',
+      labelSecondaryButton: 'Test Secondary Button',
+      isPrimaryButtonDisabled: true,
+      isSecondaryButtonDisabled: true,
+    });
+
+    expect(
+      screen.getByText('Test Primary Button').closest('button')
```

Comment:
Have you tried using `getByLabelText`?

```suggestion
      screen.getByLabelText('Test Primary Button')
```

Reply:
Updated [b07d121](https://github.com/commercetools/merchant-center-application-kit/pull/2745/commits/b07d12156c58ccca52e61f8c3e689d28689acf1c)